tests/influence/torch/test_batch_operation.py

```diff
@@ -0,0 +1,309 @@
+from dataclasses import astuple
+
+import pytest
+import torch
+
+from pydvl.influence.torch.base import TorchBatch
+from pydvl.influence.torch.batch_operation import (
+    GaussNewtonBatchOperation,
+    HessianBatchOperation,
+    InverseHarmonicMeanBatchOperation,
+)
+from pydvl.influence.torch.util import align_structure, flatten_dimensions
+
+from .test_util import model_data, test_parameters, torch
+
+
+@pytest.mark.torch
+@pytest.mark.parametrize(
+    "model_data, tol",
+    [(astuple(tp.model_params), 1e-5) for tp in test_parameters],
+    indirect=["model_data"],
+)
+def test_hessian_batch_operation(model_data, tol: float, pytorch_seed):
+    torch_model, x, y, vec, h_analytical = model_data
+
+    params = {k: p.detach() for k, p in torch_model.named_parameters()}
+
+    hessian_op = HessianBatchOperation(
+        torch_model, torch.nn.functional.mse_loss, restrict_to=params
+    )
+    batch_size = 10
+    rand_mat_dict = {k: torch.randn(batch_size, *t.shape) for k, t in params.items()}
+    flat_rand_mat = flatten_dimensions(rand_mat_dict.values(), shape=(batch_size, -1))
+    hvp_autograd_mat_dict = hessian_op.apply_to_dict(TorchBatch(x, y), rand_mat_dict)
+
+    hvp_autograd = hessian_op.apply(TorchBatch(x, y), vec)
+    hvp_autograd_dict = hessian_op.apply_to_dict(
+        TorchBatch(x, y), align_structure(params, vec)
+    )
+    hvp_autograd_dict_flat = flatten_dimensions(hvp_autograd_dict.values())
+
+    assert torch.allclose(hvp_autograd, h_analytical @ vec, rtol=tol)
+    assert torch.allclose(hvp_autograd_dict_flat, h_analytical @ vec, rtol=tol)
+
+    op_then_flat = flatten_dimensions(
```

Comment:
This test is quite long and if one of the early asserts fails you don't even get to the second ones. Maybe we could split the assertions that make sure the order of operations does not matter into a separate test?

Reply:
Agree, will split the test to make it more readable and group tests by semantics.